### PR TITLE
adds API endpoints to get merge commits logs and adds cherry-pick page 

### PR DESCRIPTION
**Pull request by rohit07sprinklr**
adds API endpoints to get merge commits logs and adds cherry-pick page 
## Previously, users used a popup to send cherry pick request. The issues associated are:
- Users did not get a option to choose specific merge commits to cherry pick. 
- There was no way to skip Cherry pick that causes conflict or an Empty cherry pick 
## Solution
- A separate page rather than a popup
- Users get a merge commit form which lists all the merge commits according to query of user.
- Checkbox for every merge commit that can be used to skip a merge commit in a cherry pick request.

### CODE REVIEW

File: src/server/merge.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+const git = require("simple-git");
+
+function wait(millis) {
```
Comment (Aditya7298) [inline src/server/merge.ts]: make a utils file and add this there. Always write DRY code.

File: src/server/getMergeCommits.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const git = require("simple-git");
+
+async function getMergeCommits(req, res, config) {
+  try {
+    const { commitAuthor, commitBranch, commitTime, location } = req.body;
+    const path = config.repos.find((repo) =>
```
Comment (Aditya7298) [inline src/server/getMergeCommits.ts:6]: extract this to a function and add in utils file. Then reuse every where.

File: src/extension/public/index.html [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
   <head>
-    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
-    <link rel="stylesheet" href="style.css">
+    <link
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css"
+      rel="stylesheet"
+      integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC"
+      crossorigin="anonymous"
+    />
+    <style>
```
Comment (Aditya7298) [inline src/extension/public/index.html:10]: why is inline css required ? Can this not be moved to a CSS file ?

File: src/extension/script.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
+import { fetchStream, streamBody } from "./fetchStream";
```
Comment (Aditya7298) [inline src/extension/script.ts:1]: This file is related to form. Should be called cherryPickForm.ts

File: src/extension/script.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+}
+function renderForm(commits, path, commitBranch, targetBranch) {
+  setContentInDesc(`${commits.length} Merge Commits Found!`);
+  if (!commits.length) {
+    return;
+  }
+  const form = document.createElement("form");
+  form.classList.add("commit-form");
+  const tableDiv = document.createElement("div");
+  tableDiv.style.height = "350px";
```
Comment (Aditya7298) [inline src/extension/script.ts]: maxHeight

File: src/extension/script.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
+import { fetchStream, streamBody } from "./fetchStream";
+
+import { getCurrentTab } from "./utils";
+
+async function cherryPickCommits(path, commitBranch, targetBranch) {
+  const table = document.querySelector(".table");
```
Comment (Aditya7298) [inline src/extension/script.ts:6]: cherryPickCommits should also receive the commits list.

File: src/extension/script.ts [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+const main = () => {
+  const cherryPickForm = document.querySelector(".cherry-pick-form");
+  cherryPickForm.addEventListener("submit", async (e) => {
+    e.preventDefault();
+    const currentTab = await getCurrentTab();
+    const formData = new FormData(e.target);
+    const jsonFormdata = {};
+    formData.forEach((value, key) => (jsonFormdata[key] = value));
```
Comment (Aditya7298) [inline src/extension/script.ts]: use reduce not forEach

File: src/extension/script.ts [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+  el.style.display = "block";
+  el.textContent = content;
+}
+function disableButton() {
+  const buttons = document.querySelectorAll("button");
+  buttons.forEach((button) => {
+    button.setAttribute("disabled", "true");
+  });
+}
+function enableButton() {
```
Comment (Aditya7298) [inline src/extension/script.ts]: enableFormButtons

File: src/extension/script.ts [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+  const buttons = document.querySelectorAll("button");
+  buttons.forEach((button) => {
+    button.removeAttribute("disabled");
+  });
+}
+function renderFormElement(commit) {
```
Comment (Aditya7298) [inline src/extension/script.ts]: which element ? Choose function and variables name carefully. Currently they do not convey what they are doing or that they represent. Check this everywhere.

File: src/extension/public/index.html (hunk 3 above)
Comment (rohit07sprinklr) [reply src/extension/public/index.html:10]: There were only few lines of style code.

File: src/extension/script.ts (hunk 6 above)
Comment (rohit07sprinklr) [reply src/extension/script.ts:6]: This will be changed in the next PR

File: src/server/getMergeCommits.ts (hunk 2 above)
Comment (rohit07sprinklr) [reply src/server/getMergeCommits.ts:6]: This function has been modified after adding option page. Will make the required changes there!

**Pull request merged**